tests/unittests/test_electrolyzer_n_inputs.py

```diff
@@ -0,0 +1,379 @@
+    database.add_data("ep2", "cost", ConstantData(20))
+
+    database.add_data("gd", "demand", ConstantData(100))
+    database.add_data("gp", "p_max", ConstantData(30))
+    database.add_data("gp", "cost", ConstantData(15))
+
+    network = Network("test")
+    network.add_node(elec_node_1)
+    network.add_node(elec_node_2)
+    network.add_node(gaz_node)
+    network.add_component(electric_prod_1)
+    network.add_component(electric_prod_2)
+    network.add_component(gaz_prod)
+    network.add_component(gaz_demand)
+    network.add_component(electrolyzer)
+    network.add_component(consumption_electrolyzer)
+
+    network.connect(
+        PortRef(electric_prod_1, "balance_port"), PortRef(elec_node_1, "balance_port")
+    )
+    network.connect(
+        PortRef(elec_node_1, "balance_port"),
+        PortRef(consumption_electrolyzer, "FlowDI1"),
+    )
+    network.connect(
+        PortRef(electric_prod_2, "balance_port"), PortRef(elec_node_2, "balance_port")
+    )
+    network.connect(
+        PortRef(elec_node_2, "balance_port"),
+        PortRef(consumption_electrolyzer, "FlowDI2"),
+    )
+    network.connect(
+        PortRef(consumption_electrolyzer, "FlowDO"), PortRef(electrolyzer, "FlowDI")
+    )
+    network.connect(PortRef(electrolyzer, "FlowDO"), PortRef(gaz_node, "balance_port"))
+    network.connect(
+        PortRef(gaz_node, "balance_port"), PortRef(gaz_demand, "balance_port")
+    )
+    network.connect(
+        PortRef(gaz_prod, "balance_port"), PortRef(gaz_node, "balance_port")
+    )
+
+    scenarios = 1
+    problem = build_problem(network, database, TimeBlock(1, [0]), scenarios)
+    status = problem.solver.Solve()
+
+    output = OutputValues(problem)
+    ep1_gen = output.component("ep1").var("generation").value
+    ep2_gen = output.component("ep2").var("generation").value
+    gp_gen = output.component("gp").var("generation").value
+
+    assert math.isclose(ep1_gen, 70)
+    assert math.isclose(ep2_gen, 30)
+    assert math.isclose(gp_gen, 30)
+
+    assert status == problem.solver.OPTIMAL
+    assert math.isclose(problem.solver.Objective().Value(), 1750)
+
+
+def test_electrolyzer_n_inputs_4():
+    """
+    Test with one electrolyzer with one input that takes every inputs
+
+    flow_ep1 + flow_ep2 * alpha_ez + flow_gp
+
+    with the same values this one is infeasible since the electrolyzer model as in the drawio doesn't work with two inputs
+    """
+    elec_node_1 = Node(model=NODE_BALANCE_MODEL, id="e1")
+    elec_node_2 = Node(model=NODE_BALANCE_MODEL, id="e2")
+    gaz_node = Node(model=NODE_BALANCE_MODEL, id="g")
+
+    electric_prod_1 = create_component(model=GENERATOR_MODEL, id="ep1")
+    electric_prod_2 = create_component(model=GENERATOR_MODEL, id="ep2")
+
+    gaz_prod = create_component(model=GENERATOR_MODEL, id="gp")
+    gaz_demand = create_component(model=DEMAND_MODEL, id="gd")
+
+    electrolyzer = create_component(model=CONVERTOR_MODEL, id="ez")
```

Comment:
I think that the problem is feasible with model CONVERTOR_MODEL_MOD instead of CONVERTOR_MODEL

Reply:
it's the same problem, since there isn't two inputs port on the electrolyzer, it does not add the two inputs and only take one